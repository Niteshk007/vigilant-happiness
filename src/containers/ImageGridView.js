import { View, FlatList } from "react-native";
import React from "react";
import ImageComponent from "../components/ImageComponent";

const imageList = [
    {
        id: "IMG001",
        src: require("../assets/images/sample.png")
    },
    {
        id: "IMG002",
        src: require("../assets/images/sample.png")
    },
    {
        id: "IMG003",
        src: require("../assets/images/sample.png")
    },
    {
        id: "IMG004",
        src: require("../assets/images/sample.png")
    },
    {
        id: "IMG005",
        src: require("../assets/images/sample.png")
    },
    {
        id: "IMG006",
        src: require("../assets/images/sample.png")
    }
]


const renderItem = ({ item, index }) => {
    return (
        <ImageComponent
            src={item?.src}
            imageStyle={{height:100,width:"100%"}}
        />
    )
}

const ImageGridView = (props) => {
    const { images } = props;
    return (
        <View style={{ flex: 1 }}>
            <FlatList
                numColumns={2}
                data={images ? images : imageList}
                renderItem={renderItem}
                keyExtractor={item => item.id}
                
            />
        </View>
    )
}

export default ImageGridView;