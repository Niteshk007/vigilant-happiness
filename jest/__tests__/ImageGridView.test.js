import React from 'react'
import { render } from "@testing-library/react-native";
import ImageGridView from "../../src/containers/ImageGridView";


describe('Should render ImageGridView with images blank array', () => {
    const props = {
        images: []
    }
    const { toJSON } = render(
        <ImageGridView {...props} />
    )
    it('Should match Snapshot', () => {
        expect(toJSON()).toMatchSnapshot();
    });
})

describe('Should render ImageGridView with images array length >0', () => {
    const props = {
        images: [{}, {}]
    }
    const { toJSON } = render(
        <ImageGridView {...props} />
    )
    it('Should match Snapshot', () => {
        expect(toJSON()).toMatchSnapshot();
    });
})

describe('Should render ImageGridView with no images props', () => {
    const { toJSON } = render(
        <ImageGridView />
    )
    it('Should match Snapshot', () => {
        expect(toJSON()).toMatchSnapshot();
    });
})