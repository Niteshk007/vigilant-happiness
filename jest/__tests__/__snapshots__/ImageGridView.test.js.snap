// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ImageGridView with images array length >0 Should match Snapshot 1`] = `null`;

exports[`Should render ImageGridView with images blank array Should match Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Should render ImageGridView with no images props Should match Snapshot 1`] = `null`;
